roadmap/implementers-guide/src/pvf-prechecking.md

```diff
@@ -1,12 +1,34 @@
 # PVF Pre-checking Overview
 
-> ⚠️ This discusses a mechanism that is currently not under-development. Follow the progress under [#3211].
+> ⚠️ This discusses a mechanism that is currently under-development. Follow the progress under [#3211].
+
+## Terms
+
+This functionality involves several processes which may be potentially
+confusing:
+
+- **Prechecking:** This is the process of initially checking the PVF when it is
+  first added. We attempt *preparation* of the PVF and make sure it succeeds
+  within a given timeout.
+- **Execution:** This actually executes the PVF. The node may not have the
+  artifact from prechecking, in which case this process also includes a
+  *preparation* job. The timeout for preparation here is more lenient than when
+  prechecking.
+- **Preparation:** This is the process of preparing the WASM blob and includes
+  both *prevalidation* and *compilation*. As prevalidation is pretty minimal
+  right now, preparation mostly consists of compilation. Note that *prechecking*
+  just consists of preparation, whereas *execution* will also prepare the PVF if
+  the artifact is not already found.
+- **Prevalidation:** Right now this just tries to deserialize the binary with
+  parity-wasm. It is a part of *preparation*.
+- **Compilation:** This is the process of compiling a PVF from wasm code to
+  machine code. It is a part of *preparation*.
```

Comment:
Nit: This book already has a glossary. Do you think it would be better to move those there? Here, we can leave a note saying that this is a loaded document with terms refer to the glossary.

Alternatively (or better, additionally) we could embed those into the text as explainations. I think as a bonus this would allow us to structure the explaination hierarchically, IMO better. Something like the following abstract:

1. In order to make the PVF usable for candidate validation it has to be registered on-chain
1. As part of the registration process, it has to go through pre-checking.
1. Pre-checking is a game of attempting preparation and reporting the results back on-chain.
1. We define preparation as a process that: validates the consistency of the wasm binary (aka prevalidation) and the compilation of the wasm module into machine code (refered to as artifact).
1. Besides pre-checking, preparation can also be triggered by execution, since compiled artifact is needed for the execution